### PR TITLE
DOCS-2511 Update Envoy tracing config example yaml to 1.19

### PR DESCRIPTION
**Pull request by kayayarai**
### What does this PR do?
Updates the sample yaml in the instructions for configuring tracing for an Envoy proxy, some general cleanup of the Envoy instructions

### Motivation
DOCS-2511 (Hotjar customer feedback)

### Preview

https://docs-staging.datadoghq.com/kari/docs-2511-envoy-update/tracing/setup_overview/proxy_setup/?tab=envoy

### Additional Notes
Checking with Eng to see if we can update the Envoy-to-C++-library version table

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/tracing/setup_overview/proxy_setup/_index.md [hunk 1]
```diff
@@ -53,66 +53,66 @@ Datadog APM is included in Envoy v1.9.0 and newer.
...
-**Note**: The example configuration below is for Envoy v1.14.
-Example configurations for older versions can be found [here][1]
+**Note**: The example configuration below is for Envoy v1.19.
+Example configurations for other versions can be found [in the `dd-opentracing-cpp` GitHub repo][1].
 
-Three settings are required to enable Datadog APM in Envoy:
+The following settings are required to enable Datadog APM in Envoy:
 
 - a cluster for submitting traces to the Datadog Agent
 - `tracing` configuration to enable the Datadog APM extension
```
Comment (cgilmour) [inline content/en/tracing/setup_overview/proxy_setup/_index.md:62]: This isn't needed anymore. Tracing is now configured in the `http_connection_manager` section.

File: content/en/tracing/setup_overview/proxy_setup/_index.md [hunk 2]
```diff
@@ -53,66 +53,66 @@ Datadog APM is included in Envoy v1.9.0 and newer.
...
+      type: strict_dns
+      lb_policy: round_robin
+      load_assignment:
+        cluster_name: datadog_agent
+        endpoints:
+        - lb_endpoints:
+          - endpoint:
+              address:
+                socket_address:
+                  address: dd-agent
```
Comment (cgilmour) [inline content/en/tracing/setup_overview/proxy_setup/_index.md:81]: The address should probably be `localhost`. An address like `dd-agent` only works in a specific environment, such as a docker-compose setup with a container named `dd-agent`.

File: content/en/tracing/setup_overview/proxy_setup/_index.md [hunk 3]
```diff
@@ -53,66 +53,66 @@ Datadog APM is included in Envoy v1.9.0 and newer.
...
-        typed_config:
-          "@type": type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager
-          tracing: {}
-```
+   ```yaml
+    - name: envoy.filters.network.http_connection_manager
+      typed_config:
+        "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
+        generate_request_id: true
+        tracing: {}
```
Comment (cgilmour) [inline content/en/tracing/setup_overview/proxy_setup/_index.md:108]: This section now needs configuring with the tracing provider and other details (same as the full example further down).

File: content/en/tracing/setup_overview/proxy_setup/_index.md [hunk 4]
```diff
@@ -124,11 +124,26 @@ static_resources:
...
-          tracing: {}
+          tracing:
+          # Use the datadog tracer
+            provider:
+              name: envoy.tracers.datadog
+              typed_config:
+                "@type": type.googleapis.com/envoy.config.trace.v3.DatadogConfig
+                collector_cluster: datadog_agent   # matched against the named cluster
+                service_name: envoy-v1.19          # user-defined service name
+          access_log:
```
Comment (cgilmour) [inline content/en/tracing/setup_overview/proxy_setup/_index.md:139]: This part is about integrating tracing and logs, is optional and could be done in different ways (eg: a JSON-based format). Not sure where that should be explained, but something to think about.
Also I think the logs pipeline for envoy needs changing to understand the additional trace and parent IDs in this format.

File: content/en/tracing/setup_overview/proxy_setup/_index.md [hunk 5]
```diff
@@ -182,18 +196,9 @@ static_resources:
         - endpoint:
             address:
               socket_address:
-                address: localhost
+                address: dd-agent
```
Comment (cgilmour) [inline content/en/tracing/setup_overview/proxy_setup/_index.md:199]: Prefer `localhost` here also.

**Review by cgilmour — APPROVED** [review]
This looks good! I'll make similar updates to the examples in dd-opentracing-cpp for the `request_id_extension` parts.

**Review by alai97 — APPROVED** [review]

**Pull request merged**